### DISCUSSION TITLE
Error composer update

### DISCUSSION THREAD

**Discussion by gradecost2** [Laravel-Backpack/community-forum#1242] [Q&A (Help)]
Hello. 
When do "composer update" we heve error  https://prnt.sc/ILjEGEGIshn_

Downloading sebastian/code-unit-reverse-lookup (3.0.0)
  0/32 [>---------------------------]   0%    Failed to download backpack/editable-columns from dist: The "https://backpackforlaravel.com/satis/download/dist/backpack/editable-columns/backpack-editable-columns-8adddb801c15871abc466bc59dbc2234664c8fab-zip-1f6c5d.zip" file could not be downloaded (HTTP/2 402 )
    Now trying to download from source
  - Syncing backpack/editable-columns (3.0.11) into cache
  1/32 [>---------------------------]   3%    Failed to download backpack/pro from dist: The "https://backpackforlaravel.com/satis/download/dist/backpack/pro/backpack-pro-b8e5537f7fa7a4a4a6fd38014fb09c25f0594fbc-zip-b145a2.zip" file could not be downloaded (HTTP/2 402 )
    Now trying to download from source
  - Syncing backpack/pro (2.2.25) into cache
 32/32 [============================] 100%
In Git.php line 473:

  Failed to execute git clone --mirror -- https://ghp...Cd5:x-oauth-basic@github.com/Laravel-Backpack/editable-columns.git d:\openserver2\modules
  \php\PHP_8.3\..\..\..\userdata\composer\cache/vcs/git-github.com-Laravel-Backpack-editable-columns.git/

  Cloning into bare repository 'd:\openserver2\modules\php\PHP_8.3\..\..\..\userdata\composer\cache/vcs/git-github.com-Laravel-Backpack-editable-
  columns.git'...
  remote: Repository not found.
  fatal: repository 'https://github.com/Laravel-Backpack/editable-columns.git/' not found

Please help.

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment]
Hey @gradecost2 

Let's try again with the following steps:
1. Visit https://backpackforlaravel.com/user/tokens ;
2. Go to the instructions tab and copy the command;
3. Run the command. It will create an `auth.json` file with the backpack access credentials;
4. Now, run `composer install` or `composer require backpack/package-you-purchased` to download the packages.

**Reply by karandatwani92** [reply]
If the error continues,  please check if you have access to the latest version. You can check that in the `packages` and `access rules` tab.

**Reply by Uteev** [reply]
I have the same problem while installing the pro. We have bought pro for multiple sites in Jun 15, 2023. So in the account I can see that  I can update until 2024-06-15. But I am not going to update. I just try to install to my new project but it does not work.

**Reply by pxpm** [reply]
Hey @Uteev the error here is related with editable columns addon. 

If your issue is with pro, please open a new discussion with your `composer.json` (atleast the backpack packages, you can ommit others) and a copy of the error your are getting 👍

**Reply by Uteev** [reply]
My problem is resolved at the moment. I was able to install pro. I had to add exact version of pro into composer.json. Previously I had in my composer.json ^2.0 but I was eligible up to 2..2.4. So composer try to update until the latest version and this caused the error. 

I think instructions could say this more clear.

**Reply by pxpm** [reply]
Thanks for the feedback @Uteev , I will open an issue in the backpack website repository to make it more clear that you should use the exact version you have access in case your license is expired. 

Cheers

**Reply by gabrielfigueroa** [reply]
https://backpackforlaravel.com/satis/download/dist/backpack/editable-columns/backpack-editable-columns-8adddb801c15871abc466bc59dbc2234664c8fab-zip-1f6c5d.zip I cant seem to access this file so I can't install backpack/pro correctly.

**Reply by gabrielfigueroa** [reply]
It's always the same package.

Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 126 installs, 0 updates, 0 removals
  - Downloading voku/portable-ascii (2.0.3)
  - Downloading symfony/polyfill-php80 (v1.31.0)
  - Downloading symfony/polyfill-mbstring (v1.31.0)
  - Downloading symfony/polyfill-ctype (v1.31.0)
  - Downloading phpoption/phpoption (1.9.3)
  - Downloading graham-campbell/result-type (v1.1.3)
  - Downloading vlucas/phpdotenv (v5.6.1)
  - Downloading symfony/css-selector (v6.4.13)
  - Downloading tijsverkoyen/css-to-inline-styles (v2.3.0)
  - Downloading symfony/deprecation-contracts (v3.5.1)
  - Downloading symfony/var-dumper (v6.4.15)
  - Downloading symfony/polyfill-uuid (v1.31.0)
  - Downloading symfony/uid (v6.4.13)
  - Downloading symfony/routing (v6.4.16)
  - Downloading symfony/process (v6.4.15)
  - Downloading symfony/polyfill-intl-normalizer (v1.31.0)
  - Downloading symfony/polyfill-intl-idn (v1.31.0)
  - Downloading symfony/mime (v6.4.17)
  - Downloading psr/container (2.0.2)
  - Downloading symfony/service-contracts (v3.5.1)
  - Downloading psr/event-dispatcher (1.0.0)
  - Downloading symfony/event-dispatcher-contracts (v3.5.1)
  - Downloading symfony/event-dispatcher (v6.4.13)
  - Downloading psr/log (3.0.2)
  - Downloading doctrine/lexer (3.0.1)
  - Downloading egulias/email-validator (4.0.3)
  - Downloading symfony/mailer (v6.4.13)
  - Downloading symfony/polyfill-php83 (v1.31.0)
  - Downloading symfony/http-foundation (v6.4.16)
  - Downloading symfony/error-handler (v6.4.17)
  - Downloading symfony/http-kernel (v6.4.17)
  - Downloading symfony/finder (v6.4.17)
  - Downloading symfony/polyfill-intl-grapheme (v1.31.0)
  - Downloading symfony/string (v6.4.15)
  - Downloading symfony/console (v6.4.17)
  - Downloading ramsey/collection (2.0.0)
  - Downloading brick/math (0.12.1)
  - Downloading ramsey/uuid (4.7.6)
  - Downloading psr/simple-cache (3.0.0)
  - Downloading nunomaduro/termwind (v1.17.0)
  - Downloading symfony/translation-contracts (v3.5.1)
  - Downloading symfony/translation (v6.4.13)
  - Downloading psr/clock (1.0.0)
  - Downloading carbonphp/carbon-doctrine-types (2.1.0)
  - Downloading nesbot/carbon (2.72.6)
  - Downloading monolog/monolog (3.8.1)
  - Downloading league/mime-type-detection (1.16.0)
  - Downloading league/flysystem (3.29.1)
  - Downloading league/flysystem-local (3.29.0)
  - Downloading nette/utils (v4.0.5)
  - Downloading nette/schema (v1.3.2)
  - Downloading dflydev/dot-access-data (v3.0.3)
  - Downloading league/config (v1.2.0)
  - Downloading league/commonmark (2.6.1)
  - Downloading laravel/serializable-closure (v1.3.7)
  - Downloading laravel/framework (v10.48.25)
  - Downloading laravel/prompts (v0.1.25)
  - Downloading guzzlehttp/uri-template (v1.0.3)
  - Downloading fruitcake/php-cors (v1.3.0)
  - Downloading webmozart/assert (1.11.0)
  - Downloading dragonmantank/cron-expression (v3.4.0)
  - Downloading doctrine/inflector (2.0.10)
  - Downloading psr/http-message (2.0)
  - Downloading psr/http-client (1.0.3)
  - Downloading ralouphie/getallheaders (3.0.3)
  - Downloading psr/http-factory (1.1.0)
  - Downloading guzzlehttp/psr7 (2.7.0)
  - Downloading guzzlehttp/promises (2.0.4)
  - Downloading guzzlehttp/guzzle (7.9.2)
  - Downloading backpack/basset (1.3.6)
  - Downloading prologue/alerts (1.2.0)
  - Downloading psr/cache (3.0.0)
  - Downloading doctrine/event-manager (2.0.1)
  - Downloading doctrine/deprecations (1.1.4)
  - Downloading doctrine/cache (2.2.0)
  - Downloading doctrine/dbal (3.9.3)
  - Downloading creativeorange/gravatar (v1.0.24)
  - Downloading backpack/crud (6.7.45)
  - Downloading backpack/editable-columns (3.0.11)
  - Downloading backpack/generators (v4.0.7)
  - Downloading backpack/theme-coreuiv4 (1.1.5)
  - Downloading fakerphp/faker (v1.24.1)
  - Downloading laravel/pint (v1.19.0)
  - Downloading symfony/yaml (v6.4.13)
  - Downloading laravel/sail (v1.39.1)
  - Downloading laravel/sanctum (v3.3.3)
  - Downloading nikic/php-parser (v5.4.0)
  - Downloading psy/psysh (v0.12.7)
  - Downloading laravel/tinker (v2.10.0)
  - Downloading hamcrest/hamcrest-php (v2.0.1)
  - Downloading mockery/mockery (1.6.12)
  - Downloading filp/whoops (2.16.0)
  - Downloading nunomaduro/collision (v7.11.0)
  - Downloading sebastian/version (4.0.1)
  - Downloading sebastian/type (4.0.0)
  - Downloading sebastian/recursion-context (5.0.0)
  - Downloading sebastian/object-reflector (3.0.0)
  - Downloading sebastian/object-enumerator (5.0.0)
  - Downloading sebastian/global-state (6.0.2)
  - Downloading sebastian/exporter (5.1.2)
  - Downloading sebastian/environment (6.1.0)
  - Downloading sebastian/diff (5.1.1)
  - Downloading sebastian/comparator (5.0.3)
  - Downloading sebastian/code-unit (2.0.0)
  - Downloading sebastian/cli-parser (2.0.1)
  - Downloading phpunit/php-timer (6.0.0)
  - Downloading phpunit/php-text-template (3.0.1)
  - Downloading phpunit/php-invoker (4.0.0)
  - Downloading phpunit/php-file-iterator (4.1.0)
  - Downloading theseer/tokenizer (1.2.3)
  - Downloading sebastian/lines-of-code (2.0.2)
  - Downloading sebastian/complexity (3.2.0)
  - Downloading sebastian/code-unit-reverse-lookup (3.0.0)
  - Downloading phpunit/php-code-coverage (10.1.16)
  - Downloading phar-io/version (3.2.1)
  - Downloading phar-io/manifest (2.0.4)
  - Downloading myclabs/deep-copy (1.12.1)
  - Downloading phpunit/phpunit (10.5.40)
  - Downloading spatie/error-solutions (1.1.2)
  - Downloading spatie/backtrace (1.7.1)
  - Downloading spatie/flare-client-php (1.10.0)
  - Downloading spatie/ignition (1.15.0)
  - Downloading spatie/laravel-ignition (2.9.0)
  74/123 [================>-----------]  60%    Failed to download backpack/editable-columns from dist: The "https://backpackforlaravel.com/satis/download/dist/backpack/editable-columns/backpack-editable-columns-8adddb801c15871abc466bc59dbc2234664c8fab-zip-1f6c5d.zip" file could not be downloaded (HTTP/2 400 )
    Now trying to download from source
  - Syncing backpack/editable-columns (3.0.11) into cache
 123/123 [============================] 100%
In Git.php line 587:

  Failed to execute git clone --mirror -- https://ghp...jvz:x-oauth-basic@github.com/Laravel-Backpack/editable-column
  s.git C:/Users/gabri/AppData/Local/Composer/vcs/git-github.com-Laravel-Backpack-editable-columns.git/

  Cloning into bare repository 'C:/Users/gabri/AppData/Local/Composer/vcs/git-github.com-Laravel-Backpack-editable-co
  lumns.git'...
  remote: Repository not found.
  fatal: repository 'https://github.com/Laravel-Backpack/editable-columns.git/' not found

**Reply by pxpm** [reply]
Hey @gabrielfigueroa please check on our website, in your tokens page, the last version you have access to each package. In your case, editable columns. Then, in your `composer.json`, make sure you use the exact version it says on the website, eg: if in the website the last version you have access to is `1.2.3`, you should set in your composer.json `backpack/editable-columns: 1.2.3`. If you use `^1.2.3` or any other "allowed" versions it will fail if you don't have access to that package version. 

Cheers
